tests/test_api_users.py

```diff
@@ -323,4 +352,196 @@ def test_03_create_update_delete_unicode_user(self):
+        set_policy("custom_attr3", scope=SCOPE.ADMIN,
+                   action="{0!s}=:*: on off".format(ACTION.SET_USER_ATTRIBUTES))
+        set_policy("custom_attr4", scope=SCOPE.ADMIN,
+                   action="{0!s}=*".format(ACTION.DELETE_USER_ATTRIBUTES))
+        with self.app.test_request_context('/user/editable_attributes/',
+                                           method='GET',
+                                           data={"user": "cornelius@realm1"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            value = result.get("value")
+            self.assertIn("delete", value)
+            self.assertEqual(value.get("delete"), ['*'])
+            self.assertIn("set", value)
+            setables = value.get("set")
+            self.assertIn("*", setables)
+            self.assertIn("hello", setables)
+            self.assertIn("hello2", setables)
+            self.assertEqual(["on", "off"], setables.get("*"))
+            self.assertEqual(["one", "two", "three"], setables.get("hello"))
+            self.assertEqual(["*"], setables.get("hello2"))
+
+        set_policy("custom_create_user", scope=SCOPE.ADMIN,
+                   action=ACTION.ADDUSER)
+        # CREATE a user
+        self._create_user_wordy()
+        self._get_wordy_auth_token()
+        # The admin sets attributes for this user:
+        with self.app.test_request_context('/user/attribute',
+                                           method='POST',
+                                           data={"user": "wordy@sqlrealm",
+                                                 "key": "hello",
+                                                 "value": "one"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertTrue(result.get("status"), res.data)
+            self.assertTrue(result.get("value") >= 0)
+
+        # We let the user wordy@sqlrealm login and GET his attributes
+        with self.app.test_request_context('/user/attribute',
+                                           method='GET',
+                                           data={"user": "cornelius@realm1"},
+                                           headers={'Authorization': self.wordy_auth_token}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertTrue(result.get("status"), res.data)
+            self.assertIn("hello", result.get("value"))
+            self.assertEqual("one", result.get("value").get("hello"))
+
```

Comment:
Please also add a test that the user is not able to set attribute himself

Reply:
Will do.

Reply:
Done in 421376b3